cli/version/version_match.go

```diff
@@ -0,0 +1,156 @@
+package version
+
+import (
+	"fmt"
+
+	"github.com/tarantool/tt/cli/util"
+)
+
+const (
+	IgnoredVersionPart = ^uint64(0)
+)
+
+type CompareResult uint16
+
+const (
+	// Values according order of Fields in regex from `matchVersionParts()`
+	CompareNotMatch CompareResult = iota
+	CompareMatchBuildName
+	CompareMatchMajor
+	CompareMatchMinor
+	CompareMatchPatch
+	CompareMatchReleaseType
+	CompareMatchReleaseNum
+	CompareMatchAdditional
+	CompareMatchHash
+	CompareMatchRevision
+
+	CompareMatchTotal = CompareMatchRevision
+)
+
+type MatchVersion struct {
+	Version
+	requiredFields []CompareResult
+}
+
+func makeEmptyMatchVersion() MatchVersion {
+	return MatchVersion{
+		Version: Version{
+			Major:      IgnoredVersionPart,
+			Minor:      IgnoredVersionPart,
+			Patch:      IgnoredVersionPart,
+			Additional: IgnoredVersionPart,
+			Revision:   IgnoredVersionPart,
+			Release: Release{
+				Type: TypeRelease,
+				Num:  IgnoredVersionPart,
+			},
+		},
+		requiredFields: make([]CompareResult, 0, CompareMatchTotal),
+	}
+}
+
+func Explore(verStr string) (MatchVersion, error) {
+	version := makeEmptyMatchVersion()
+
+	matches, err := matchVersionParts(verStr, false)
+	if err != nil {
+		return version, err
+	}
+
+	version.BuildName = matches["buildname"]
+	if version.BuildName != "" {
+		version.requiredFields = append(version.requiredFields, CompareMatchBuildName)
+	}
+
+	if matches["major"] != "" {
+		if version.Major, err = util.AtoiUint64(matches["major"]); err != nil {
+			return version, fmt.Errorf("can't parse Major; %s", err)
```

Comment:
The usual error wrapping looks like this:
```suggestion
			return version, fmt.Errorf("can't parse Major: %w", err)
```